cluster/src/main/java/org/apache/iotdb/cluster/entity/raft/DataStateMachine.java

```diff
@@ -82,59 +76,77 @@ public DataStateMachine(String groupId, PeerId peerId) {
   @Override
   public void onApply(Iterator iterator) {
     while (iterator.hasNext()) {
-
-      Closure closure = null;
-      DataGroupNonQueryRequest request = null;
-      BasicResponse response = null;
+      final Closure closure = iterator.done();
       final ByteBuffer data = iterator.getData();
+
+//      /** It's leader to apply task **/
+//      if (closure != null) {
+//        RaftTaskManager.getInstance().execute(() -> applySingleTask(closure, data));
+//      } else {
+        applySingleTask(closure, data);
+//      }
+      iterator.next();
+    }
+  }
+
+  /**
+   * Apply a single raft task. If this node is the leader of state machine's data group, apply the
```

Comment:
`If this node is the leader of state machine's data group, apply the...` this statement is useless?

Reply:
yes, it has been removed.